Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+#  **.NET Core Installer Task V1**
+
+The task installs user specified version of .NET Core SDK/Runtime. This can be consumed to supply a particular version of .Net Core SDK/runtime to the subsequent tasks in pipeline.
+
+### **A new major version of the task is created because of the following breaking changes:**
+- **Installating multiple sdk/runtime versions side by side:** Users had asked for the feature where multiple versions of sdk/runtime can be installed and then be used in subsequent tasks in a pipeline.(one project may require multiple .Net Core sdks to build different applications), for more information about user's ask, [refer here](https://github.com/Microsoft/azure-pipelines-tasks/issues/8306). This is a breaking change, as previously the V0 task always provided only one version of sdk/runtime.
+
+## New features added in V1 task:
+
+## **- Support for fetching latest version for a given major or major.minor version**
+This feature was being asked by users who wanted to install the latest version as soon as it becomes available, without manually changing the version in the task. For information about user's asks, refer [here](https://github.com/Microsoft/azure-pipelines-tasks/issues/9171).
+
+This feature is implemented by supporting patterns in version input: `version`. This input now accepts .Net sdk/runtime versions in the following patterns
+- `Major.Minor.PatchVersion :` complete version as present in releases.json for the Major.Minor Version channel. Example: 2.2.104
+- `Major.Minor.x :` Latest version in the Major.Minor version channel release. It can also be the latest preview version if `includePreviewVersions` input is enabled/true. This can be used to get all latest bug fixes automatically in a major.minor version. As back-compat is maintained in major.minor version, thus ensuring applications being built will not break.
+- `Major.x :` Latest version released in the Major version channel. It can also be the latest preview version if `includePreviewVersions` input is enabled/true. This can be used by tools or library developers who need to test with every latest version for compatibility.
+
+### How it works:
+The correct version needs to be identified in case user enters pattern based version such as 2.2.x. This is how the exact needed version and its information is extracted:
+
+- [releases-index.json](https://github.com/dotnet/core/blob/master/release-notes/releases-index.json) The file containing links to all .Net Core release channels is download and read into a JSON object `releasesIndex`. This is used to find all available release channels and links to their releases.json.
+
+- The `version` input is divided into three sections:
+  - Major
+  - Minor
+  - Patch
+
+- Based on Minor versions value, the following can happen
+  - Numeric : we get the corresponding channel's (Major.Minor) releases.json link from `releasesIndex`. Then the releases.json is downloaded and parsed into an object: `channelInformation`.
```

Comment:
What is difference between channelInformation and releaseInformation - both seem to be for release.json

Reply:
channelInformation is releases-index.json 
releaseinformation is releases.json